packages/protocol-kit/src/utils/passkeys/PasskeySigner.ts

```diff
@@ -0,0 +1,238 @@
+  }
+
+  static async init(
+    passkey: passkeyArgType,
+    safeWebAuthnSignerFactoryContract: SafeWebAuthnSignerFactoryContractImplementationType,
+    provider: Provider
+  ): Promise<PasskeySigner> {
+    const coordinates = await extractPasskeyCoordinates(passkey.publicKey)
+
+    return new PasskeySigner(
+      passkey.rawId,
+      coordinates,
+      safeWebAuthnSignerFactoryContract,
+      provider
+    )
+  }
+
+  /**
+   * Returns the address associated with the Signer.
+   * @returns {Promise<string>} A promise that resolves to the signer's address.
+   */
+  async getAddress(): Promise<string> {
+    const [signerAddress] = await this.safeWebAuthnSignerFactoryContract.getSigner([
+      BigInt(this.coordinates.x),
+      BigInt(this.coordinates.y),
+      BigInt(P256_VERIFIER_ADDRESS)
+    ])
+
+    return signerAddress
+  }
+
+  /**
+   * Encodes the createSigner contract function.
+   * @returns {string} The encoded data to create a signer.
+   */
+  encodeCreateSigner(): string {
+    return this.safeWebAuthnSignerFactoryContract.encode('createSigner', [
+      BigInt(this.coordinates.x),
+      BigInt(this.coordinates.y),
+      BigInt(P256_VERIFIER_ADDRESS)
+    ])
+  }
+
+  /**
+   * Signs the provided data using the passkey.
+   * @param {Uint8Array} data - The data to be signed.
+   * @returns {Promise<string>} A promise that resolves to the signed data.
+   */
+  async sign(data: Uint8Array): Promise<string> {
+    const assertion = (await navigator.credentials.get({
+      publicKey: {
+        challenge: data,
+        allowCredentials: [{ type: 'public-key', id: this.passkeyRawId }],
+        userVerification: 'required'
+      }
+    })) as PublicKeyCredential & { response: AuthenticatorAssertionResponse }
+
+    if (!assertion || !assertion?.response?.authenticatorData) {
+      throw new Error('Failed to sign data with passkey Signer')
+    }
+
+    const { authenticatorData, signature, clientDataJSON } = assertion.response
+
+    return ethers.AbiCoder.defaultAbiCoder().encode(
+      ['bytes', 'bytes', 'uint256[2]'],
+      [
+        new Uint8Array(authenticatorData),
+        extractClientDataFields(clientDataJSON),
+        extractSignature(signature)
+      ]
+    )
+  }
+
+  connect(provider: Provider): ethers.Signer {
+    return new PasskeySigner(
+      this.passkeyRawId,
+      this.coordinates,
+      this.safeWebAuthnSignerFactoryContract,
+      provider
+    )
+  }
+
+  signTransaction(): Promise<string> {
+    throw new Error('Passkey Signers cannot sign transactions, they can only sign data.')
+  }
+
+  signMessage(message: string | Uint8Array): Promise<string> {
+    if (typeof message === 'string') {
+      return this.sign(ethers.getBytes(message))
+    }
+
+    return this.sign(message)
+  }
+
+  signTypedData(): Promise<string> {
+    throw new Error('Passkey Signers cannot sign signTypedData, they can only sign data.')
```

Comment:
Consider on changing the `isTypedDataSigner` as well (although that will probably change the error).